Feature: Configurable solver

fmitchell-r7 opened the pull request:
This fixes #84. The solver used by Berkshelf is now configurable. Users can choose between the `:gecode` and `:ruby` solvers in their Berksfile. We default to the `:gecode` solver to preserve the old behavior of explicitly using that solver.

athompson-r7 commented:
LGTM

dgreene-r7 commented on template/Berksfile.erb:
```diff
@@ -8,3 +8,5 @@ source '<%= s %>'
 cookbook '<%= name %>', '<%= cookbook.fetch(:version, '>= 0.0.0') %>', <%= cookbook.to_hash %>
 <% end -%>
+
+solver :<%= berkshelf.solver %>
```
Do you know if `ERB` does a `to_s` on symbols before exposing them to the template? I genuinely don't know, but if it doesn't your output will be `solver ::gecode` (or whatever).

fmitchell-r7 replied:
It doesn't. Anything between `<%=` and `%>` winds up being spit out as a string. So the symbol `:ruby` becomes the string `ruby` and we have to prefix it with a colon for Berkshelf.

jmanero-r7 commented on lib/builderator/interface/berkshelf.rb:
```diff
@@ -33,6 +33,10 @@ def berkshelf_config
         directory.join('Berksfile')
       end
+
+      def solver
```
You shouldn't need this for the template. `<%= berkshelf.solver %>` actually reference the Config object directly now.

fmitchell-r7 commented:
@jmanero-r7, I changed the test to check for a default value in the Config instead of the Berkshelf interface and removed the accessor for the solver from the Berkshelf interface.

jmanero-r7 commented:
👍 Looks good
